part2_server.py

```diff
@@ -1,43 +1,33 @@
 # Group#: G37
 # Student Names: Ryley McRae & Niko Andrianos
 
+from socket import *
+import random, time
+
 """
-    This program implements a server for a UDP Pinger 
-    application on a local host.
+    Implements a server for a UDP Pinger application on a local host.
 """
+class Server:
+    def __init__(self, host: str = "127.0.0.1", port: int = 12000, bufferSize: int = 2048):
+        self.host = host
+        self.port = port
+        self.bufferSize = bufferSize
+        self.serverSocket = socket(AF_INET, SOCK_DGRAM)
+
+    def run(self):
+        self.serverSocket.bind((self.host, self.port))
+        print(f"The server is listening on port {self.port}")
+        while True:
+            message, clientAddress = self.serverSocket.recvfrom(self.bufferSize)
+            ignoreMessageFlag = (random.random() < 0.1) # create a flag with a 10% probability to simulate packet loss
```

Comment:
Personally, I think:
`ignoreMessageFlag = True if random.random() < 0.1 else False`
is better for readability.

Reply:
I prefer this for readability too. Hopefully it would not be considered redundant code